apps/mark/frontend/src/api.ts

```diff
@@ -63,6 +71,34 @@ export const getAuthStatus = {
   },
 } as const;
 
+export const getElectionDefinitionFromCard = {
+  queryKey(): QueryKey {
+    return ['getElectionDefinitionFromCard'];
+  },
+  useQuery(
+    electionHash?: string,
+    options: UseQueryOptions<Result<Optional<ElectionDefinition>, Error>> = {}
+  ) {
+    const apiClient = useApiClient();
+    return useQuery(
+      this.queryKey(),
+      async () => {
+        const result = await apiClient.readElectionDefinitionFromCard({
+          electionHash,
+        });
+        const electionData = result.ok();
+        const electionDefinition = electionData
+          ? safeParseElectionDefinition(electionData).ok()
+          : undefined;
+        return ok<Optional<ElectionDefinition>, Error>(electionDefinition);
```

Comment:
I moved this election definition parsing logic into the API client to avoid having to duplicate it, though I suppose I could move it even further upstream to the backend itself, in the spirit of avoiding substantial logic in the API client. Can go ahead and do that

Reply:
And done: https://github.com/votingworks/vxsuite/pull/3011/commits/fd8ed5b2cfd79c0139f7181302fc04da7da198b7